### PR TITLE
Add `DerivativeFilter`

### PR DESCRIPTION
**Pull request by fhagemann**
This PR addresses https://github.com/JuliaPhysics/RadiationDetectorDSP.jl/pull/52.
Hopefully, it gives the desired results.

### CODE REVIEW

File: src/derivative.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Base.@kwdef struct DerivativeFilter{T <: RadiationDetectorDSP.RealQuantity} <: AbstractRadIIRFilter
+    "Filter gain"
+    gain::T = 1
+end
+export DerivativeFilter
+
+struct DerivativeFilterInstance{T <: RadiationDetectorDSP.RealQuantity, TG <: RadiationDetectorDSP.RealQuantity} <: AbstractRadSigFilterInstance{LinearFiltering}
+    gain::TG
+    n_input::Int
+end
```
Comment (fhagemann) [inline src/derivative.jl:33]: I'm not sure that the parametric types (`T` and `TG`) are needed for, I just put them there.

File: src/derivative.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+function rdfilt!(y::AbstractVector, fi::DerivativeFilterInstance{T}, x::AbstractVector
+    ) where T
+
+    @assert firstindex(y) == firstindex(x)
+    @assert lastindex(y) == lastindex(x)
+    @inbounds @simd for i in eachindex(y)
+        y[i] = fi.gain * (x[max(i,begin+1)] - x[max(i-1,begin)])
+    end
+    y
+end
```
Comment (fhagemann) [inline src/derivative.jl:56]: `T` is not used in the `rdfilt!` method. Is that wanted?

Comment (fhagemann) [reply src/derivative.jl]: --> removed in 37b444c

File: src/LegendDSP.jl [hunk 3]
```diff
@@ -49,6 +49,7 @@ include("ml.jl")
 include("dsp_ml_routines.jl")
 include("multi_intersect.jl")
 include("moving_window_multi.jl")
+include("derivative.jl")
```
Comment (theHenks) [inline src/LegendDSP.jl:52]: I need this filter in all `dsp_routines` so could you move this up and include all fitlers first?

**Review by theHenks — CHANGES_REQUESTED** [review]
Looks great, thanks @fhagemann! 
I just need the `include` first to use it in the `dsp` routines.

File: src/LegendDSP.jl (hunk 3 above)
Comment (fhagemann) [reply src/LegendDSP.jl]: Done!

File: src/derivative.jl [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    gain::TG
+    n_input::Int
+end
+
+function RadiationDetectorDSP.fltinstance(flt::DerivativeFilter{TG}, si::SamplingInfo{T}) where {T, TG}
+    DerivativeFilterInstance{T, TG}(flt.gain, _smpllen(si))
+end
+
+
+RadiationDetectorDSP._filterlen(fi::DerivativeFilterInstance) = 1
```
Comment (apmypb) [inline src/derivative.jl:42]: Shouldn't it be two? Or does it have to be one because `rdfilt` uses it to construct the correct output length?

**Review by apmypb — APPROVED** [review]

**Review by theHenks — APPROVED** [review]

**Pull request merged**